plenum/server/node.py

```diff
@@ -1389,22 +1393,33 @@ def checkInstances(self) -> None:
         logger.debug("{} choosing to start election on the basis of count {} and nodes {}".
                      format(self, self.connectedNodeCount, self.nodestack.conns))
 
-    def adjustReplicas(self):
+    def adjustReplicas(self,
+                       old_required_number_of_instances: int,
+                       new_required_number_of_instances: int):
         """
         Add or remove replicas depending on `f`
         """
         # TODO: refactor this
         newReplicas = 0
-        while len(self.replicas) < self.requiredNumberOfInstances:
-            self.replicas.grow()
+        while old_required_number_of_instances < new_required_number_of_instances:
+            self.replicas.add_replica(old_required_number_of_instances)
             newReplicas += 1
-            self.processStashedMsgsForReplica(len(self.replicas) - 1)
-        while len(self.replicas) > self.requiredNumberOfInstances:
-            self.replicas.shrink()
+            self.processStashedMsgsForReplica(old_required_number_of_instances)
+            old_required_number_of_instances += 1
```

Comment:
The parameter name does not suppose the value changes.

Reply:
Done